### PR TITLE
[WIP][CI] Add check-project workflow

### PR DESCRIPTION
**Pull request by gamesh411**
Create a workflow for analyzing Java projects via PR versions of
Spotbugs. Baseline diff and time measurement features are added to
monitor the impact of PRs (result set changes or possible regressions).

----

Make sure these boxes are checked before submitting your PR -- thank you!

- [x] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

**Comment by gamesh411** [comment]
Hi!
I would like to suggest an enhancement to the CI workflow of checking PR-s.
This patch is big (because of the binary content of the projects to be analyzed), and I would like to discuss whether it is better to rely on just downloading the projects.

- The upside of downloading is that the repo does not need to contain binary content.
- The downside of downloading is that we cannot easily control the analysis target inside the PR itself.

Aside from that, any suggestions are appreciated.
Cheers!

File: .github/workflows/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Check projects with Spotbugs Github Action
+
+[Github Actions](https://docs.github.com/en/actions)  can be used to automate CI/CD activities.
+A series of such action is called a workflow.
```
Comment (ThrawnCA) [inline .github/workflows/README.md:4]: "action" -> "actions"

File: .github/workflows/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+This repository contains 2 workflows which can be used to automatically run a Spotbugs Java projects.
```
Comment (ThrawnCA) [inline .github/workflows/README.md:5]: "projects" -> "project"

File: .github/workflows/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+The [check-projects](check-projects.yml) workflow builds a Spotbugs runnable from master (labelled baseline) and the current PR (labelled new).
+The unit tests are *not* run in this job on purpose to keep it simple and allow for a fast response time.
+After building the baseline and new versions of Spotbugs, an analysis is run on open source projects.
+The list of projects currently analyzed:
+  - spotbugs
+  - matsim-libs
+  - jenkins
+
+The workflow analyses these projcts via the baseline and the new versions of Spotbugs and produces a diff of analysis results for each one.
```
Comment (ThrawnCA) [inline .github/workflows/README.md:20]: "projcts" -> "projects"

File: .github/workflows/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+## Installation
+
+These workflows are intended to be installed on a Spotbugs fork.
+On the default branch of the repository (main or master by default) the workflow description files should be placed into the `.github/workflows` directory.
+Both [check-projects.yml](check-projects.yml) and [check-projects-comment.yml](check-projects-comment.yml) files should be there and must be committed.
+
+
+## Usage
+
+Once installed you can use the workflows by creating a feature branch in you fork, and creating a pull-request from that branch to your forks master branch.
```
Comment (ThrawnCA) [inline .github/workflows/README.md:33]: you fork -> your fork

your forks -> your fork's

File: .github/workflows/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+The outcome of the analysis is seen on the job status of the PR, and the diff of results can be seen as PR comment.
```
Comment (ThrawnCA) [inline .github/workflows/README.md:34]: diff of results -> diff of the results

as PR comment -> as a PR comment

**Review by ThrawnCA — CHANGES_REQUESTED** [review]

**Review by KengoTODA — COMMENTED** [review]
I prefer avoiding binary file committed in the Git repo. Please consider to remove .jar files from this PR.

File: .github/workflows/README.md (hunk 2 above)
Comment (KengoTODA) [reply .github/workflows/README.md:5]: Actually we already has workflows in `.github/workflows` so "This repository contains 2 workflows" sounds strange. Is it OK to move these explanation into YAML file itself?

**Comment by KengoTODA** [comment]
I'll close this inactive PR. Please feel free to reopen if necessary.

**Pull request closed without merging**